stratified-sampling/script.whizzml

```diff
@@ -0,0 +1,41 @@
+;; Takes a dataset id and a catagorical field id, and breaks the
+;; dataset into pieces, one per value. Returns the list of datasets.
+
+(define (split-it ds-id fld-id)
+  (let (bucket-list (map (lambda (x) (head x)) 
+                         ((fetch ds-id) 
+                             ["fields" fld-id "summary" "categories"])))
+    (iterate (ds-list []            
+              bucket bucket-list)
+      (let (split-filter (flatline "(= (f {{fld-id}}) {{bucket}})")
+            split-ds (create-dataset {"origin_dataset" ds-id
+                                      "lisp_filter" split-filter}))
+        (append ds-list split-ds)))))
+
+;; Takes a list of datasets and a list of counts, and randomly samples
```

Comment:
nit: it actually takes a list of dataset ids (also, returns a list of dataset ids)